### PR TITLE
Add: add note feature v1.0b

### PR DESCRIPTION
**Pull request by ong-ck**

### CODE REVIEW

File: src/main/java/seedu/clialgo/TopicManager.java [hunk 1]
```diff
@@ -25,6 +28,32 @@ public TopicManager() {
...
+     * Name of the note to be added cannot be the same as a previously added note.
+     *
+     * @param noteName Name of the note file.
+     * @param topicName Name of the topic the note is added to.
+     * @param note The <code>Note</code> object representing the note file.
+     * @return True if note is successfully added and False otherwise.
+     */
+    public boolean addNote(String noteName, String topicName, Note note) {
+        // Check if note name has been taken
+        if (allNotes.contains(noteName)) {
```
Comment (heejet) [inline src/main/java/seedu/clialgo/TopicManager.java:44]: I have added a new method isRepeatedNote in TopicManager which could be used here.

File: src/main/java/seedu/clialgo/command/AddCommand.java [hunk 2]
```diff
@@ -2,46 +2,67 @@
...
+         */
+
+        // Check if topicName is valid
+        if (!topicManager.isValidTopic(topic)) {
+            new InvalidTopicCommand(topic).execute(topicManager, ui, fileManager);
+            return;
+        }
 
-        // Check if added -> execute invalid command if note added
+        boolean isAdded = topicManager.addNote(name, topic, newNote);
```
Comment (heejet) [inline src/main/java/seedu/clialgo/command/AddCommand.java:56]: Would it be better to rename `isAdded` to `isSuccessfullyAdded`

**Review by heejet — COMMENTED** [review]
Good Job!

File: src/main/java/seedu/clialgo/Topic.java [hunk 3]
```diff
@@ -1,19 +1,37 @@
...
+     * @param topicName The name of the topic.
+     */
     public Topic(String topicName) {
         this.topicName = topicName;
         notes = new HashMap<>();
     }
 
-    public boolean addNote(String name, String path) {
-        return true;
+    public HashMap<String, Note> getNotes() {
```
Comment (nicholas132000) [inline src/main/java/seedu/clialgo/Topic.java:23]: Good use of getter method to access private attributes from outside this class

**Review by lohjooh — COMMENTED** [review]
LGTM!

**Pull request merged**